web-common/src/features/dashboards/proto-state/toProto.ts

```diff
@@ -278,20 +278,41 @@ function toPivotProto(pivotState: PivotState): PartialMessage<DashboardState> {
     };
   return {
     pivotIsActive: true,
+    pivotRowDimensions: pivotState.rows.dimension.map((d) => {
+      if (d.type === PivotChipType.Dimension) {
```

Comment:
Would be good to break the mapper out reduce the complexity here, similar to how we do in `fromProto`